crow/src/lib.rs

```diff
@@ -128,8 +136,23 @@ fn to_topup_info(topup: ListAvailableTopupsTopup) -> graphql::Result<TopupInfo>
         "The backend returned an incomplete topup - missing lnurlw",
     )?;
 
+    let status = match topup.status {
+        topup_status_enum::FAILED => Some(TopupStatus::FAILED),
+        topup_status_enum::READY => Some(TopupStatus::READY),
+        topup_status_enum::SETTLED => Some(TopupStatus::SETTLED),
+        _ => None,
+    }
+    .ok_or_runtime_error(
+        graphql::GraphQlRuntimeErrorCode::CorruptData,
+        format!(
+            "The backend returned an unkown topup status - unkown status {:?}",
+            topup.status
+        ),
+    )?;
```

Comment:
On top of that, I would probably be more specific with the _other_ arm: Instead of `_ => {},` I'd write `topup_status_enum::Other(_) =>{},` .

The reason for this is to move some errors into compile time rather than runtime: If the GraphQL schema were to change and another Enum case was added, the project wouldn't compile anymore because that new enum case would be missing from the match statement.

If we just do `_ => {}`, then it would compile just fine and we would potentially forget to add the newly added enum, getting errors at runtime later on.